src/posix/mm.rs

```diff
@@ -0,0 +1,99 @@
+//! Memory management
+
+use types::{int_t, uint_t, ulong_t, void_t, size_t, intptr_t};
+use types::{off_t};
+use types::{rlimit};
+#[cfg(all(target_os = "macos", target_arch = "x86_64"))]
+use types::{caddr_t};
+
+use syscalls::{sys_getrlimit, sys_mmap, sys_munmap};
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+use syscalls::{sys_brk};
+
+use rust::prelude::*;
+
+use consts::errno::{ENOMEM};
+use libc::errno::{errno};
+
+macro_rules! forward {
+    ($sys:ident, $($p:expr),*) => {
+        match $sys($($p),*) {
+            n if n < 0 => {
+                errno = -n;
+                -1
+            },
+            n => n,
+        }
+    };
+}
+
+/// Increases the data break to the given address, returning 0 on success
+/// or -1 on failure, setting errno to ENOMEM.
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+#[no_mangle]
+pub unsafe extern fn brk(addr: *const void_t) -> int_t {
+    match sys_brk(addr as ulong_t) == 0 {
+        true => 0,
+        false => {
+            errno = ENOMEM;
+            -1
+        }
+    }
+}
+
+/// Increments the data break by `increment`, returning either the previous
+/// break or `((void*)-1)` on failure, setting errno to ENOMEM.
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+#[no_mangle]
+pub unsafe extern fn sbrk(increment: intptr_t) -> *const void_t {
+    let oldbrk: *const void_t = sys_brk(0) as *const void_t;
+    if increment != 0 {
+        let newbrk: *const void_t = offset(oldbrk, increment as isize);
```

Comment:
The problem is many interfaces (including `brk()`) accept a `const void*`, so we still need to handle pointer arithmetic `*const void_t` correctly.

Since our `void_t` is a `u8`, we should be safe.